rockcraft/extensions/_utils.py

```diff
@@ -64,7 +64,7 @@ def _apply_extension(
     if "parts" not in yaml_data:
         yaml_data["parts"] = {}
 
-    parts: dict[str, Any] = yaml_data["parts"]
+    parts = cast(dict[str, Any], yaml_data["parts"])
```

Comment:
It doesn't really matter, but quoting the first value passed to `cast` will let us keep any related imports in `if CHECKING:` blocks

```suggestion
    parts = cast("dict[str, Any]", yaml_data["parts"])
```